packages/shared/src/components/auth/OnboardingRegistrationForm.tsx

```diff
@@ -0,0 +1,155 @@
+import React, {
+  ReactElement,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import { useMutation } from 'react-query';
+import { checkKratosEmail } from '../../lib/kratos';
+import { AuthFormProps, providerMap } from './common';
+import OrDivider from './OrDivider';
+import AnalyticsContext from '../../contexts/AnalyticsContext';
+import { AuthEventNames, AuthTriggersOrString } from '../../lib/auth';
+import { ExperimentWinner } from '../../lib/featureValues';
+import { Button } from '../buttons/Button';
+import AuthForm from './AuthForm';
+import { TextField } from '../fields/TextField';
+import MailIcon from '../icons/Mail';
+import { IconSize } from '../Icon';
+import Alert, { AlertParagraph, AlertType } from '../widgets/Alert';
+
+interface OnboardingRegistrationFormProps extends AuthFormProps {
+  onExistingEmail?: (email: string) => unknown;
+  onSignup?: (email: string) => unknown;
+  onProviderClick?: (provider: string, login?: boolean) => unknown;
+  targetId?: string;
+  isLoginFlow?: boolean;
+  logInTitle?: string;
+  signUpTitle?: string;
+  trigger: AuthTriggersOrString;
+  isReady: boolean;
+}
+
+const OnboardingRegistrationForm = ({
+  onSignup,
+  onExistingEmail,
+  onProviderClick,
+  targetId,
+  isReady,
+  trigger,
+}: OnboardingRegistrationFormProps): ReactElement => {
+  const { trackEvent } = useContext(AnalyticsContext);
+  const [shouldLogin, setShouldLogin] = useState(false);
+  const [registerEmail, setRegisterEmail] = useState<string>(null);
+  const { mutateAsync: checkEmail } = useMutation((emailParam: string) =>
+    checkKratosEmail(emailParam),
+  );
+
+  useEffect(() => {
+    trackEvent({
+      event_name: shouldLogin
+        ? AuthEventNames.OpenLogin
+        : AuthEventNames.OpenSignup,
+      extra: JSON.stringify({ trigger }),
+      target_id: targetId || ExperimentWinner.AuthVersion,
+    });
+    // @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [shouldLogin]);
+
+  const getFormEmail = useCallback((e: React.FormEvent) => {
+    const form = e.currentTarget as HTMLFormElement;
+    const input = Array.from(form.elements).find(
+      (el) => el.getAttribute('name') === 'email',
+    ) as HTMLInputElement;
+
+    return input?.value?.trim();
+  }, []);
+
+  const onEmailSignup = async (e: React.FormEvent) => {
+    e.preventDefault();
```

Comment:
Maybe also do a early return if `checkEmail` mutation is already loading?

Reply:
Now with early return